compiler/docgen.nim

```diff
@@ -26,6 +26,16 @@ const
 
 type
   TSections = array[TSymKind, Rope]
+  Example = ref object
+    ## a single runnableExamples
+    rdoccmd: string
+  ExampleGroup = ref object
+    ## a group of runnableExamples with same rdoccmd
+    rdoccmd: string ## from 1st arg in `runnableExamples(rdoccmd): body`
+    docCmd: string ## from user config, eg --doccmd:-d:foo
+    code: string ## contains imports; each import contains `body`
+    index: int ## group index
+    examples: seq[Example]
```

Comment:
```suggestion
    examples: seq[string]
```
and remove the `Example` type

Reply:
done to unblock this PR, but I preferred the way I wrote it as it's more self documenting and I intended to build upon `Example` in future PR; `compiler/docgen.nim` deals with `runnableExamples` so it's natural for it to have a type representing a runnableExample